File: c++/7_strukturData/Enum.cpp
#include <iostream>
using namespace std;

enum warna {merah, kuning, hijau = 10, biru};

int main(){

    //membuat instance enum
	warna baju;

	baju = merah;   //maka hasilnya 0
    baju = kuning;  //maka hasilnya 1
	baju = hijau;   //maka hasilnya 10
	baju = biru;    //maka hasilnya 11

    cout << baju << endl;

	cin.get();
	return 0;
}

/*
enum bisa dibilang koleksi kata yang sebenarnya nilainya adalah index dan akan terus melanjutkan indexnya ditambah 1 secara urut pada nilai sebelumnya
sehingga jika hijau diberi index 10 maka biru akan melanjutkan indexnya menjadi 11

meskipun memberikan semua koleksi kata nilai index. Kita hanya bisa menggunakan satu kata berindex untuk digunakan;
*/

File: java/7_strukturData/Enum.java

class Enum {
    
    public static void main(String[] args) {
        

        System.out.println(warna.merah);
        System.out.println(warna.hijau);
        System.out.println(warna.biru);
    }

    static enum warna{merah, kuning, hijau, biru};
}

/*
 * enum di c++ dan java aga berbeda. Meskipun sama sama berfungsi untuk mengkoleksi data konstan
 * 
 * di java, index dari koleksi data enum tidak bisa dirubah. Ini sudah paten dimulai dari 0.
 * Dan nilai dari koleksi enum warna otomatis terinisialisasi string nilai didalam body{}
 * Pelajari lagi enum;
 */